### PR TITLE
feat: update aws agent guides

### PR DESCRIPTION
**Pull request by paulbalaji**
resolves https://github.com/hyperlane-xyz/issues/issues/963

- Updates existing validator guide to be more AWS-focused, pointing back to the local agents guide where required.
- Updates existing relayer guide to be in line with updated validator guide. Add callout to top of relayer guide, encouraging readers to understand local setup + AWS validator setup first.
- Add a new deploy-with-terraform guide, with diagrams explaining the terraform example code.

Quick access:
- [validators guide](https://hyp-v3-docs-6vb7dqkey-abacus-works.vercel.app/docs/operate/validators/run-validators)
- [relayer guide](https://hyp-v3-docs-6vb7dqkey-abacus-works.vercel.app/docs/operate/relayer/run-relayer)
- [terraform guide](https://hyp-v3-docs-6vb7dqkey-abacus-works.vercel.app/docs/operate/deploy-with-terraform)

TODO:
- [x] update link to monorepo after merging https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/3170

### CODE REVIEW

**Comment by paulbalaji** [comment]
Quick access:
- [validators guide](https://hyp-v3-docs-6vb7dqkey-abacus-works.vercel.app/docs/operate/validators/run-validators)
- [relayer guide](https://hyp-v3-docs-6vb7dqkey-abacus-works.vercel.app/docs/operate/relayer/run-relayer)
- [terraform guide](https://hyp-v3-docs-6vb7dqkey-abacus-works.vercel.app/docs/operate/deploy-with-terraform)

**Comment by avious00** [comment]
@nambrot @tkporter friendly ping :)

File: docs/operate/deploy-with-terraform.mdx [hunk 1]
```diff
@@ -0,0 +1,207 @@
+# Deploy with Terraform
+
+For those more familiar with deploying to AWS through infrastructure-as-code tools such as Terraform, we provide an example configuration designed to set up the necessary infrastructure for a Hyperlane Validator on AWS. It automates the creation of resources such as ECS clusters, VPCs, subnets, and security groups required for running a Validator node.
+
+[**Click here to navigate to the example Terraform configuration.**](https://github.com/hyperlane-xyz/hyperlane-monorepo/tree/main/rust/terraform)
```
Comment (paulbalaji) [inline docs/operate/deploy-with-terraform.mdx:5]: PR merged, so link has been updated  to `main` branch

File: docs/operate/deploy-with-terraform.mdx [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+    ECSTask -- Signs with --> KMSKey
+    ECSTask -- Perms from --> IAMUser
+
+```
+
+## Usage
+
+As a prerequisite, you need to have Terraform installed and the AWS CLI configured with your credentials.
+
+To initialise the terraform state:
```
Comment (tkporter) [inline docs/operate/deploy-with-terraform.mdx:68]: we speak AMERICAN here 🇺🇸 🇺🇸 🦅 🦅 🦅

File: docs/operate/deploy-with-terraform.mdx [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+    Internet[Internet]
+    IGW -- Connects to --> Internet
+    EIP -- Connects to --> Internet
+```
+
+## Known issues
+
+### Non-core configuration
+
+It is non-trivial to set custom configuration values for non-core chains.
```
Comment (tkporter) [inline docs/operate/deploy-with-terraform.mdx:198]: do you think "non-core chain" is obvious to someone new to Hyperlane? wondering if "a chain you've deployed Hyperlane to yourself" or something is more palatable. Don't feel strongly tho, happy to leave as is too

File: docs/operate/relayer/run-relayer.mdx [hunk 4]
```diff
@@ -30,147 +36,189 @@ flowchart TB
...
+Then start the container with the relevant arguments. For example, your configuration for AWS:
 
 ```sh
 docker run \
   -it \
-  --mount type=bind,source="$(pwd)"/hyperlane_db,target=/hyperlane_db \
-  $DOCKER_IMAGE \
-  ./relayer --db /hyperlane_db --relayChains ethereum,polygon,avalanche
+  -e AWS_ACCESS_KEY_ID=ABCDEFGHIJKLMNOP \
+  -e AWS_SECRET_ACCESS_KEY=xX-haha-nice-try-Xx \
```
Comment (tkporter) [inline docs/operate/relayer/run-relayer.mdx:188]: hahah

**Review by tkporter — APPROVED** [review]
love it! great job, very thorough and easy to follow

File: docs/operate/deploy-with-terraform.mdx [hunk 5]
```diff
@@ -0,0 +1,207 @@
+# Deploy with Terraform
+
+For those more familiar with deploying to AWS through infrastructure-as-code tools such as Terraform, we provide an example configuration designed to set up the necessary infrastructure for a Hyperlane Validator on AWS. It automates the creation of resources such as ECS clusters, VPCs, subnets, and security groups required for running a Validator node.
```
[the remark this answers is not in the document]
Comment (paulbalaji) [reply docs/operate/deploy-with-terraform.mdx:3]:
```suggestion
For those more familiar with deploying to AWS through infrastructure-as-code tools such as Terraform, we provide an example configuration designed to set up the necessary infrastructure for a Hyperlane Validator on AWS. It automates the creation of resources such as ECS clusters, VPCs, subnets, and security groups required for running a Validator agent.
```

File: docs/operate/deploy-with-terraform.mdx (hunk 2 above)
Comment (paulbalaji) [reply docs/operate/deploy-with-terraform.mdx:68]:
```suggestion
To initialize the terraform state:
```

File: docs/operate/relayer/run-relayer.mdx (hunk 4 above)
Comment (paulbalaji) [reply docs/operate/relayer/run-relayer.mdx:188]: just doing my bit to protect us from @laurenceday 🫡

File: docs/operate/deploy-with-terraform.mdx (hunk 3 above)
Comment (paulbalaji) [reply docs/operate/deploy-with-terraform.mdx:198]: I think if you've made it to this point in the guide you should know what it is, but we've not used "non-core" as a term before, will change to your suggestion to make it more obvious

File: docs/operate/deploy-with-terraform.mdx [hunk 6]
```diff
@@ -193,9 +193,9 @@ graph TD
 
 ## Known issues
 
-### Non-core configuration
+### PI configuration
 
-It is non-trivial to set custom configuration values for non-core chains.
+It is non-trivial to set custom configuration values for PI chains, such as ones you've deployed Hyperlane to yourself.
 Currently you may have to just pass a long list of environment variables or command line arguments.
```
Comment (paulbalaji) [inline docs/operate/deploy-with-terraform.mdx:199]: @tkporter this sounding ok?

**Pull request merged**